hedera-node/src/main/java/com/hedera/services/files/SpecialFileSystem.java

```diff
@@ -0,0 +1,229 @@
+				.setRealmNum(ApplicationConstants.DEFAULT_FILE_REALM)
+				.setShardNum(ApplicationConstants.DEFAULT_FILE_SHARD).build();
+		this.fileSystemLocation = EntityIdUtils.asLiteralString(nodeAccountID);
+
+		createEmptyIfNotExist(fid150);
+		invalidateHash();
+	}
+
+	public SpecialFileSystem(HashMap<FileID, byte[]> fileMap, String accountIDStr) {
+		this.fileMap = (HashMap<FileID, byte[]>) fileMap.clone();
+		this.fileSystemLocation = accountIDStr;
+		invalidateHash();
+	}
+
+	public HashMap<FileID, byte[]> getFileMap() {
+		return fileMap;
+	}
+
+	public byte[] getFileHash(FileID fileID) {
+		byte[] fileBytes = getFileContent(fileID);
+		try {
+			return MessageDigest.getInstance("SHA-384").digest(fileBytes);
+		} catch (NoSuchAlgorithmException e) {
+			log.error("Error when get hash of file {}", fileID);
+			return null;
+		}
+	}
+
+	private void createEmptyIfNotExist(FileID fileID) {
+		File testFile = new File(SPECIAL_FILESYSTEM_DIR + fileSystemLocation + File.separator + fileIDtoDotString(fileID));
+		if (testFile.exists()) {
+			try {
+				byte[] hash = MessageDigest.getInstance("SHA-384").digest(getFileContent(fileID));
+				fileMap.put(fileID, hash);
+				log.info("File {} already exists, saving its hash in map {}", fileID, hex(hash));
+			} catch (NoSuchAlgorithmException e) {
+				log.error("Error when calculating hash of file {}", fileID);
+			}
+		} else {
+			log.info("Creating empty File {}", fileID);
+			put(fileID, new byte[0]);
+		}
+	}
+
+	private static String fileIDtoDotString(FileID fileID) {
+		return "File" + fileID.getShardNum() + "." + fileID.getRealmNum() + "." + fileID.getFileNum();
+	}
+
+	public synchronized byte[] getFileContent(FileID fileID) {
+		try {
+			return FileUtils.readFileToByteArray(
+					new File(SPECIAL_FILESYSTEM_DIR + fileSystemLocation + File.separator + fileIDtoDotString(fileID)));
+		} catch (IOException e) {
+			log.error(String.format("Error when reading fileID %s from local filesystem %s", fileID, e));
+			return new byte[0];
+		}
+	}
+
+	public synchronized void put(FileID fileID, byte[] content) {
+		try {
+			byte[] hash = MessageDigest.getInstance("SHA-384").digest(content);
+			log.info("New file size {} with hash {}", content.length, hex(hash));
+			fileMap.put(fileID, hash);
+			FileUtils.writeByteArrayToFile(
+					new File(SPECIAL_FILESYSTEM_DIR + fileSystemLocation + File.separator + fileIDtoDotString(fileID)),
+					content);
+			invalidateHash();
+		} catch (IOException e) {
+			log.error("Error when writing fileID {} to local filesystem", fileID, e);
+		} catch (NoSuchAlgorithmException e) {
+			log.error("Error when calculating hash of file {}", fileID);
+		}
+	}
+
+	public boolean isSpeicalFileID(FileID fileID) {
+		return fileMap.containsKey(fileID);
+	}
+
+	@Override
+	public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+		int mapSize = in.readInt();
+		for (int i = 0; i < mapSize; i++) {
+			long realmID = in.readLong();
```

Comment:
Mismatch with the serialization order; here we read `(realm, shard, num)` and in `serialize` write `(shard, realm, num)`. Addressed in #658.